File: backend/schemas/pipeline.py
"""流水线相关 Pydantic 模型"""
from typing import Optional, List, Dict, Any
from pydantic import BaseModel


class OutputConfig(BaseModel):
    file_output: bool = True
    db_output: bool = False
    tag_table: str = "enterprise_tag"
    fact_table: str = "enterprise_fact"


class PipelineRunRequest(BaseModel):
    enterprise_names: List[str]
    output_config: OutputConfig = OutputConfig()


class PipelineRunResponse(BaseModel):
    task_id: str
    enterprise_count: int
    status: str = "queued"


class TaskSummary(BaseModel):
    task_id: str
    enterprise_count: int
    status: str
    completed_count: int = 0
    failed_count: int = 0
    created_at: Optional[str] = None


class EnterpriseResult(BaseModel):
    enterprise_name: str
    status: str
    confidence: float = 0
    elapsed_seconds: float = 0
    token_total: int = 0
    tag_summary: List[str] = []
    fact_count: int = 0


class ProgressMessage(BaseModel):
    type: str = "progress"
    enterprise_name: str = ""
    progress: int = 0
    stage: str = ""
    step: str = ""
    detail: str = ""
    elapsed_seconds: float = 0
    data_sources: List[Dict[str, Any]] = []
    token_usage: Dict[str, Any] = {}

File: frontend/src/types/pipeline.ts
/** 流水线相关 TypeScript 类型定义 */

export interface OutputConfig {
  file_output: boolean
  db_output: boolean
  tag_table?: string
  fact_table?: string
}

export interface PipelineRunRequest {
  enterprise_names: string[]
  output_config: OutputConfig
}

export interface PipelineRunResponse {
  task_id: string
  enterprise_count: number
  status: string
}

export interface DataSourceStatus {
  platform: string
  status: 'success' | 'failed' | 'skipped' | 'running'
  pages: number
  text_length?: number
}

export interface TokenUsageStage {
  input_tokens: number
  output_tokens: number
  total_tokens: number
}

export interface TokenUsage {
  total: number
  stages: Record<string, TokenUsageStage>
}

export interface ProgressMessage {
  type: 'progress' | 'enterprise_complete' | 'batch_complete' | 'pong'
  task_id?: string
  enterprise_name?: string
  progress?: number
  stage?: string
  step?: string
  detail?: string
  elapsed_seconds?: number
  data_sources?: DataSourceStatus[]
  token_usage?: TokenUsage
  status?: string
  confidence?: number
  tag_summary?: string[]
  fact_count?: number
  token_total?: number
  error?: string
  batch_progress?: { completed: number; total: number }
  results?: EnterpriseResultSummary[]
}

export interface EnterpriseResultSummary {
  enterprise_name: string
  status: string
  confidence: number
  elapsed_seconds?: number
  token_total?: number
  tag_summary?: string[]
  fact_count?: number
}

export interface TaskInfo {
  task_id: string
  enterprise_count: number
  status: string
  completed_count: number
  failed_count: number
  created_at: string
}
